File: performance/src/main/kotlin/org/jetbrains/ring/ClassBaselineBenchmark.kt
package org.jetbrains.ring

open class ClassBaselineBenchmark {

    //Benchmark 
    fun consume() {
        for (item in 1..BENCHMARK_SIZE) {
            Blackhole.consume(Value(item))
        }
    }

    //Benchmark 
    fun consumeField() {
        val value = Value(0)
        for (item in 1..BENCHMARK_SIZE) {
            value.value = item
            Blackhole.consume(value)
        }
    }

    //Benchmark 
    fun allocateList(): List<Value> {
        val list = ArrayList<Value>(BENCHMARK_SIZE)
        return list
    }

    //Benchmark 
    fun allocateArray(): Array<Value?> {
        val list = arrayOfNulls<Value>(BENCHMARK_SIZE)
        return list
    }

    //Benchmark 
    fun allocateListAndFill(): List<Value> {
        val list = ArrayList<Value>(BENCHMARK_SIZE)
        for (item in 1..BENCHMARK_SIZE) {
            list.add(Value(item))
        }
        return list
    }

    //Benchmark 
    fun allocateListAndWrite(): List<Value> {
        val value = Value(0)
        val list = ArrayList<Value>(BENCHMARK_SIZE)
        for (item in 1..BENCHMARK_SIZE) {
            list.add(value)
        }
        return list
    }

    //Benchmark 
    fun allocateArrayAndFill(): Array<Value?> {
        val list = arrayOfNulls<Value>(BENCHMARK_SIZE)
        var index = 0
        for (item in 1..BENCHMARK_SIZE) {
            list[index++] = Value(item)
        }
        return list
    }
}

File: performance/KotlinVsSwift/ring/src/ClassBaselineBenchmark.swift
/*
 * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
 * that can be found in the LICENSE file.
 */

import Foundation

class ClassBaselineBenchmark {

    func consume() {
        for item in 1...Constants.BENCHMARK_SIZE {
            Blackhole.consume(Value(item))
        }
    }
    
    func consumeField() {
        let value = Value(0)
        for item in 1...Constants.BENCHMARK_SIZE {
            value.value = item
            Blackhole.consume(value)
        }
    }
    
    func allocateList() -> [Value] {
        let list = [Value](repeating: Value(0), count: Constants.BENCHMARK_SIZE)
        return list
    }
    
    func allocateArray() -> [Value?] {
        let list = [Value?](repeating: nil, count: Constants.BENCHMARK_SIZE)
        return list
    }
    
    func allocateListAndFill() -> [Value] {
        var list: [Value] = []
        for item in 1...Constants.BENCHMARK_SIZE {
            list.append(Value(item))
        }
        return list
    }
    
    func allocateListAndWrite() -> [Value] {
        let value = Value(0)
        var list: [Value] = []
        for _ in 1...Constants.BENCHMARK_SIZE {
            list.append(value)
        }
        return list
    }
    
    func allocateArrayAndFill() -> [Value?] {
        var list = [Value?](repeating: nil, count: Constants.BENCHMARK_SIZE)
        var index = 0
        for item in 1...Constants.BENCHMARK_SIZE {
            list[index] = Value(item)
            index += 1
        }
        return list
    }
}
